mono/metadata/reflection.c

```diff
@@ -1312,6 +1312,11 @@ method_body_object_construct (MonoDomain *domain, MonoClass *unused_class, MonoM
 	memcpy (il_data, header->code, header->code_size);
 	mono_gchandle_free_internal (il_gchandle);
 
+#ifdef ENABLE_NETCORE
+	g_assert_not_reached ();
```

Comment:
I am hitting this assert when I try to run xunit tests for System.Runtime.Tests using 
```
make run-xunit COREFX_ROOT=/prj/corefx-egor TEST_ASSEMBLY=/prj/corefx-egor/artifacts/bin/System.Runtime.Tests/netcoreapp-Unix-Debug/System.Runtime.Tests.dll
```